Uchu.StandardScripts/NinjagoMonastery/LanternsOfSpinjitzu.cs

```diff
@@ -0,0 +1,29 @@
+using Uchu.World;
+using Uchu.World.Scripting.Native;
+
+namespace Uchu.StandardScripts.NinjagoMonastery
+{
+    [ScriptName("ScriptComponent_1620_script_name__removed")]
+    public class LanternOfSpinjitzu : ObjectScript
+    {
+        /// <summary>
+        /// Creates the object script.
+        /// </summary>
+        /// <param name="gameObject">Game object to control with the script.</param>
+        public LanternOfSpinjitzu(GameObject gameObject) : base(gameObject)
+        {
+            // Listen to players interacting with railposts
+            Listen(gameObject.OnInteract, async player =>
+            {
+                if (!player.TryGetComponent<MissionInventoryComponent>(out var missionInventory))
+                    return;
+                if (!player.TryGetComponent<CharacterComponent>(out var characterComponent))
+                    return;
+
+                // If player has Cole's mission active, set flag to proceed with next mission
+                if (missionInventory.HasActive(2072))
+                    await characterComponent.SetFlagAsync(2020, true);
```

Comment:
Regarding to my previous comment with the flags. Ignore if there's no enum / you don't really mind

Reply:
It would improve readability to create an enum if we don't. There is a difference between "flag 2072" and "flag SomeMissionName".

Reply:
There is a FactionFlag enum for the 4 factions, I'll change this to not be only for factions